src/main/java/com/esri/samples/displayinformation/sketch_on_map/SketchOnMapSample.java

```diff
@@ -0,0 +1,415 @@
+
+        // if sketch is valid, enable save and stop sketch buttons
+        if (sketchEditor.isSketchValid()) {
+          saveButton.setText("Save sketch to graphics overlay");
+          saveButton.setDisable(false);
+          stopSketchButton.setDisable(false);
+        } else {saveButton.setDisable(true);}
+
+        // if the sketch editor can undo, enable the undo button otherwise disable it
+        if (sketchEditor.canUndo()) {
+          undoButton.setDisable(false);
+        } else {
+          undoButton.setDisable(true);
+        }
+
+        // if the sketch editor can redo, enable the redo button otherwise disable it
+        if (sketchEditor.canRedo()) {
+          redoButton.setDisable(false);
+        } else {
+          redoButton.setDisable(true);
+        }
+      });
+
+      // create a control panel
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0, 0, 0, 0.3)"), CornerRadii.EMPTY, Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10));
+      controlsVBox.setMaxSize(255, 110);
+      controlsVBox.getStyleClass().add("panel-region");
+
+      // create a flow pane for placing buttons side by side within the control box
+      FlowPane flowPaneUndoRedo = new FlowPane(Orientation.HORIZONTAL, 70, 10, undoButton, redoButton);
+      flowPaneUndoRedo.setAlignment(Pos.CENTER);
+      controlsVBox.getChildren().addAll(sketchComboBox, stopSketchButton, flowPaneUndoRedo, saveButton, editButton, clearButton);
+
+      // add the map view to the stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox);
+      stackPane.setAlignment(controlsVBox, Pos.TOP_RIGHT);
+      stackPane.setMargin(controlsVBox, new Insets(10, 10, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * When the done button is clicked, check that sketch is valid. If so, get the geometry from the sketch, set its
+   * symbol and add it to the graphics overlay.
+   */
+  private void storeGraphicInGraphicOverlay() {
+
+    // if the sketch isn't valid, stop the sketch editor.
+    if (!sketchEditor.isSketchValid()) {
+      sketchEditor.stop();
+      return;
+    }
+
+    Geometry sketchGeometry = sketchEditor.getGeometry();
+    // if the sketch geometry isn't null, and the graphics overlay isn't empty
+    // get the selected graphic, and set its geometry to that of the sketch editor geometry
+    // otherwise, create a new graphic based on the sketch editor geometry and give it a symbol
+    if (sketchGeometry != null) {
+      if (!graphicsOverlay.getSelectedGraphics().isEmpty()) {
+        graphic = graphicsOverlay.getSelectedGraphics().get(0);
+        graphic.setGeometry(sketchGeometry);
+      } else {
+        graphic = new Graphic(sketchGeometry);
+        if (graphic.getGeometry().getGeometryType() == GeometryType.POLYGON) {
```

Comment:
use a switch here
```java
switch (sketchGeometry.getGeometryType()) {
    case POLYGON:
        graphic.setSymbol(fillSymbol);
        break;
    case POLYLINE:
        graphic.setSymbol(lineSymbol);
        break;
    case POINT:
    case MULTIPOINT:
        graphic.setSymbol(pointSymbol);
}
```

Reply:
Another alternative is to have three graphics overlays: points, lines, and polygons. Then specify a renderer for the overlay with the appropriate symbol. The advantage is you can just create graphics with geometry and add it to the right overlay and it will inherit the symbol of the overlay's renderer.